python/spec/templates/jax/xla_ops.cpp

```diff
@@ -1,169 +1,197 @@
+// Generated JAX FFI bindings for celerite2.
+// Regenerate with: python python/spec/generate.py
+
 #include <pybind11/pybind11.h>
-#include <cstdint>
-#include <stdexcept>
+#include <Eigen/Core>
 #include <string>
-#include <type_traits>
+
+#include "xla/ffi/api/ffi.h"
+#include "xla/ffi/api/api.h"
+
 #include "../driver.hpp"
 
 namespace py = pybind11;
+namespace ffi = xla::ffi;
 using namespace celerite2::driver;
 
+// Helpers
+template <typename Buffer>
+inline Eigen::Index dim0(const Buffer& buf) {
+  return static_cast<Eigen::Index>(buf.dimensions()[0]);
+}
+template <typename Buffer>
+inline Eigen::Index dim1(const Buffer& buf) {
+  return static_cast<Eigen::Index>(buf.dimensions()[1]);
+}
```

Comment:
I think something like:

```suggestion
template <int Axis, typename Buffer>
inline Eigen::Index dim(const Buffer& buf) {
  return static_cast<Eigen::Index>(buf.dimensions()[Axis]);
}
```

might work and be cleaner, but I don't feel strongly!